### PR TITLE
Improved energy distribution using NeSST

### LINKED ISSUE #88: improved neutron energy
instead of a 14MeV or a 2.5MeV Muir distribution we could have a distribution that accounts for DD, TT and DT reactions

```python
import NeSST as nst

def create_neutron_source_term(
        ion_temperature: float,
        fuel: dict = {'D':0.5, 'T':0.5},
) -> openmc.stats.Discrete:
    """Finds the energy distribution
"""

    ion_temperature = ion_temperature / 1e3  # convert eV to keV

    sum_fuel_isotopes = sum(fuel.values())
    if sum_fuel_isotopes > 1.:
        raise ValueError(f'isotope fractions within the fuel must sum to be below 1. Not {sum_fuel_isotopes}')

    if sum_fuel_isotopes < 0.:
        raise ValueError(f'isotope must sum to be above 0. Not {sum_fuel_isotopes}')

    for k, v in fuel.dict:
        if k not in ['D', 'T']:
            raise ValueError('Fuel dictionary keys must be either "D" or "T" not "{k}".)
        if v < 0
            raise ValueError('Fuel dictionary values must be above 0 not "{k}".)
        if v > 1
            raise ValueError('Fuel dictionary values must be below 1 not "{k}".)

    #Set atomic fraction of D and T in scattering medium and source
    if 'D' in fuel.keys():
        nst.frac_D_default = fuel['D']
    else:
        nst.frac_D_default = 0

    if 'T' in fuel.keys():
        nst.frac_T_default = fuel['T']
    else:
        nst.frac_T_default = 0

    # 1.0 neutron yield, all reactions scaled by this value
    num_of_vals = 500
    # single grid for DT, DD and TT grid
    E_pspec = np.linspace(0, 20, num_of_vals)  # accepts MeV units

    dNdE_DT_DD_TT = np.zeros(num_of_vals)
    if 'D' in fuel.keys() and 'T' in fuel.keys():
        DTmean, DTvar = nst.DTprimspecmoments(ion_temperature)
        Y_DT = 1.0
        dNdE_DT = Y_DT * nst.Qb(E_pspec, DTmean, DTvar)  # Brysk shape i.e. Gaussian
        dNdE_DT_DD_TT= dNdE_DT_DD_TT + dNdE_DT
    if 'D' in fuel.keys()
        DDmean, DDvar = nst.DDprimspecmoments(ion_temperature)
        Y_DD = nst.yield_from_dt_yield_ratio("dd", 1.0, ion_temperature)
        dNdE_DD = Y_DD * nst.Qb(E_pspec, DDmean, DDvar)  # Brysk shape i.e. Gaussian
        dNdE_DT_DD_TT= dNdE_DT_DD_TT + dNdE_DD
    if 'T' in fuel.keys()
        Y_TT = nst.yield_from_dt_yield_ratio("tt", 1.0, ion_temperature)
        dNdE_TT = Y_TT * nst.dNdE_TT(E_pspec, ion_temperature)
        dNdE_DT_DD_TT= dNdE_DT_DD_TT + dNdE_TT

    return openmc.stats.Discrete(E_pspec * 1e6, dNdE_DT_DD_TT)
```

### PR DESCRIPTION
**Pull request by shimwell**
This PR is an attempt to make use of NeSST to define the neutron spectra. Closes #88 

I have a bit more work to do before it is ready, for example I'm going to see if I can combined the Muir spectra for DD and DT neutrons with the openmc.stats.Discrete distribution for the TT neutrons.

going to try combining with [this](https://docs.openmc.org/en/stable/pythonapi/generated/openmc.data.combine_distributions.html)

### CODE REVIEW

**Comment by shimwell** [comment]
This PR has turned into a bit of a big refactoring, sorry for the large diff

On the plus side it does now solve #80 and #37 and touches on addressing Liams comments in #47

**Comment by shimwell** [comment]
tests are passing locally, so are the examples.

**Comment by shimwell** [comment]
This PR is currently reducing the package by a reasonable amount of lines of code while also adding functionality.

I've redo the checking of values so that it is more concise.

I'm keen to see if [this](https://github.com/aidancrilly/NeSST/issues/22) issue can be fixed before we bring in nesst as a dependency

**Comment by shimwell** [comment]
I just noticed another part of the code that will need changing to support DD and DT reactions. It looks like this part is only considering DT reactions

https://github.com/fusion-energy/openmc-plasma-source/blob/64e4aa13f1e81efb8d0e16d7b76acb40bacce468/src/openmc_plasma_source/tokamak_source.py#L416

So we might need to add DD reactions

Also I guess the ion_density for DD pairs is also half that of DT pairs

File: pyproject.toml [hunk 1]
```diff
@@ -35,7 +36,8 @@ write_to = "src/_version.py"
 [project.optional-dependencies]
 tests = [
     "pytest>=5.4.3",
-    "hypothesis"
+    "hypothesis",
+    "NeSST>=1.0.0"
```
Comment (jon-proximafusion) [inline pyproject.toml:40]:
```suggestion
    "NeSST>=1.1.0"
```
this one makes use of endf python package for cross sections

File: src/openmc_plasma_source/fuel_types.py [hunk 2]
```diff
@@ -1,37 +1,208 @@
...
+
+        dd_source = openmc.stats.Normal(mean_value=DDmean, std_dev=DD_std_dev)
+        # normal could be done with Muir but in this case we have the mean and std dev from NeSST
+        # dd_source = openmc.stats.muir(e0=DDmean * 1e6, m_rat=4, kt=ion_temperature)
+
+        dt_source = openmc.stats.Normal(mean_value=DTmean, std_dev=DT_std_dev)
+        # normal could be done with Muir but in this case we have the mean and std dev from NeSST
+        # dt_source = openmc.stats.muir(e0=DTmean * 1e6, m_rat=5, kt=ion_temperature)
+
+        # todo look into combining distributions openmc.data.combine_distributions()
```
Comment (jon-proximafusion) [inline src/openmc_plasma_source/fuel_types.py:203]:
```suggestion
        openmc_univariate = [tt_source, dd_source, dt_source]
        probabilities = [strength_TT / total_strength, strength_DD / total_strength, strength_DT / total_strength]
        return openmc.data.combine_distributions(openmc_univariate, probabilities)
```

**Comment by shimwell** [comment]
closing in favour of #95

**Pull request closed without merging**